src/posix/platform/system.cpp

```diff
@@ -52,7 +52,16 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
     ot::Posix::RadioUrl radioUrl(aPlatformConfig->mRadioUrl);
 
 #if OPENTHREAD_POSIX_VIRTUAL_TIME
-    virtualTimeInit(static_cast<uint16_t>(atoi(radioUrl.GetValue("forkpty-arg"))));
+    // The last argument must be the node id
+    {
+        const char *nodeId = radioUrl.GetValue("forkpty-arg");
+
+        for (const char *arg = nodeId; (arg = radioUrl.GetValue("forkpty-arg", nodeId)) != nullptr; nodeId = arg)
+        {
+        }
```

Comment:
```suggestion
        const char *nodeId = radioUrl.GetValue("forkpty-arg");
        const char *arg;

        while ((arg = radioUrl.GetValue("forkpty-arg", nodeId)) != nullptr)
        {
            nodeId = arg;
        }
```

Assigning `arg = nodeId` is useless?

Reply:
```suggestion
        const char *nodeId = nullptr;

        for (const char *arg = nullptr; (arg = radioUrl.GetValue("forkpty-arg", arg)) != nullptr; nodeId = arg)
        {
        }
```